### PR TITLE
Graceful node shutdowns

### PR DESCRIPTION
**Pull request by heavenlyhash**

### CODE REVIEW

File: appliance/postgresql/runner.go [hunk 1]
```diff
@@ -198,7 +202,7 @@ func runCmd(cmd *exec.Cmd) {
 				os.Exit(status.ExitStatus())
```
Comment (titanous) [inline appliance/postgresql/runner.go]: This looks important.

File: sdutil/sdutil.go [hunk 2]
```diff
@@ -1,6 +1,11 @@
 package main
 
+import (
+	"github.com/flynn/flynn/pkg/shutdown"
+)
+
 func main() {
+	defer shutdown.Exit()
```
Comment (titanous) [inline sdutil/sdutil.go]: We already do signal handling in sdutil and it will be deleted soon, so no need.

File: test/apps/echoer/main.go [hunk 3]
```diff
@@ -15,12 +15,15 @@ func main() {
 	port := os.Getenv("PORT")
 	addr := ":" + port
 
-	if _, err := discoverd.AddServiceAndRegister(service, addr); err != nil {
+	hb, err := discoverd.AddServiceAndRegister(service, addr)
+	if err != nil {
 		log.Fatal(err)
 	}
+	defer hb.Close()
```
Comment (titanous) [inline test/apps/echoer/main.go]: Any reason not to use shutdown for these test apps?

Comment (archSeer) [inline test/apps/echoer/main.go]: In one of the commit messages: "Use the pkg/shutdown system in all flynn services.  Small test/demo/example apps just use plain defers and idiomatic go error handling, to make it perfectly clear there's nothing special about the flynn shutdown convenience package."

Comment (heavenlyhash) [inline test/apps/echoer/main.go]: They're short enough there's no compelling need, and I figured they might be used as examples.  We could convert them; wanna?

Comment (titanous) [inline test/apps/echoer/main.go]: Yeah, the example apps are fine without it but use it for the test apps.

**Comment by titanous** [comment]
LGTM

**Comment by heavenlyhash** [comment]
There's a couple more races to be addressed in the cli 'run' subcommand, then this should finally be good.

**Comment by heavenlyhash** [comment]
This should be good now.

**Comment by titanous** [comment]
LGTM

**Pull request merged**